source/server/server.cc

```diff
@@ -614,15 +614,18 @@ RunHelper::RunHelper(Instance& instance, const Options& options, Event::Dispatch
       }) {
   // Setup signals.
   if (options.signalHandlingEnabled()) {
-// TODO(Pivotal): Figure out solution to graceful shutdown on Windows. None of these signals exist
-// on Windows.
-#ifndef WIN32
+#ifdef WIN32
+    sigterm_ = dispatcher.listenForSignal(ENVOY_WIN32_SIGTERM, [&instance]() {
+      ENVOY_LOG(warn, "caught ENVOY_WIN32_SIGTERM");
+      instance.shutdown();
+    });
+#else
     sigterm_ = dispatcher.listenForSignal(SIGTERM, [&instance]() {
       ENVOY_LOG(warn, "caught SIGTERM");
       instance.shutdown();
     });
 
-    sigint_ = dispatcher.listenForSignal(SIGINT, [&instance]() {
+    sigint_ = dispatcher.listenForSignal(SIGINIT, [&instance]() {
```

Comment:
nit: typo should be `SIGINT` (no change)

Reply:
(looks like that is a compile error anyway)

Reply:
I saw that but I am on a different VM so I am waiting for a tiny bit to fix it.

Reply:
fixed